internal/events/eventer.go

```diff
@@ -103,14 +103,16 @@ type Eventer struct {
 	// webhookSubscriber will subscribe to the webhook topic and handle incoming events
 	webhookSubscriber message.Subscriber
 	// TODO: We'll have a Final publisher that will publish to the final topic
-	metrics *messageInstruments
+	msgInstruments *messageInstruments
 
 	closer driverCloser
 }
 
 type messageInstruments struct {
 	// message processing time duration histogram
 	messageProcessingTimeHistogram metric.Int64Histogram
+	// poison queue counter
+	poisonQueueCounter metric.Int64Counter
```

Comment:
Another option here is to use an attribute to record whether the message succeeded or failed, which would allow correlating latency and mesages which are discarded.  Not required, but it can help if you want to filter out "messages that were slow due to being filtered as poison".

Reply:
Thanks, addressed all comments in https://github.com/stacklok/minder/pull/2052/commits/80775056b176e550505fc5752f17304d8586188c

Reply:
Update: we now reuse the histogram metric with an added attribute if the message has been poisoned or not.
The `recordMetrics func` was also fixed as per the PR description.